### PR TITLE
Implement ApplicationSettings and ConfigurationProvider

### PR DESCRIPTION
**Pull request by p1v0zavr**

### CODE REVIEW

**Comment by p1v0zavr** [comment]
Depends on @PublicFedorenko  :)

File: FaynaTown.Tennis.Planner.Bot.Core/Providers/Secret/CurrentProcessSecretProvider.cs [hunk 1]
```diff
@@ -1,20 +1,24 @@
...
+﻿using System;
+using Ardalis.GuardClauses;
+using FaynaTown.Tennis.Planner.Bot.Core.Configuration;
 using Microsoft.Extensions.Logging;
-using System;
 
 namespace FaynaTown.Tennis.Planner.Bot.Core.Providers.Secret;
+
 internal class CurrentProcessSecretProvider : ISecretProvider {
+    private readonly IConfigurationProvider ConfigurationProvider;
```
Comment (PublicFedorenko) [inline FaynaTown.Tennis.Planner.Bot.Core/Providers/Secret/CurrentProcessSecretProvider.cs:9]:
```suggestion
    private readonly IConfiguration Configuration;
```

File: FaynaTown.Tennis.Planner.Bot.Core/Providers/Secret/CurrentProcessSecretProvider.cs [hunk 2]
```diff
@@ -1,20 +1,24 @@
...
     private readonly ILogger Logger;
 
-    public CurrentProcessSecretProvider(ILogger logger) {
+    public CurrentProcessSecretProvider(IConfigurationProvider configurationProvider, ILogger logger) {
+        ConfigurationProvider = Guard.Against.Null(configurationProvider);
```
Comment (PublicFedorenko) [inline FaynaTown.Tennis.Planner.Bot.Core/Providers/Secret/CurrentProcessSecretProvider.cs:13]:
```suggestion
    public CurrentProcessSecretProvider(IConfiguration configuration, ILogger logger) {
        Configuration = Guard.Against.Null(configuration);
```

File: FaynaTown.Tennis.Planner.Bot.Core/Providers/Secret/CurrentProcessSecretProvider.cs [hunk 3]
```diff
@@ -1,20 +1,24 @@
...
         Logger = Guard.Against.Null(logger);
     }
 
     public string? GetSecret(string secretName) {
         try {
-            return Environment.GetEnvironmentVariable(secretName);
+            return ConfigurationProvider.GetConfigurationValue(secretName);
```
Comment (PublicFedorenko) [inline FaynaTown.Tennis.Planner.Bot.Core/Providers/Secret/CurrentProcessSecretProvider.cs:19]:
```suggestion
            return Configuration[Guard.Against.NullOrWhiteSpace(secretName)] ?? default;
```

File: FaynaTown.Tennis.Planner.Bot.Azure/Factories/TelegramBotClientFactory.cs [hunk 4]
```diff
@@ -1,17 +1,20 @@
 ﻿using Ardalis.GuardClauses;
+using FaynaTown.Tennis.Planner.Bot.Core.Configuration;
 using FaynaTown.Tennis.Planner.Bot.Core.Providers.Secret;
 using Telegram.Bot;
 
 namespace FaynaTown.Tennis.Planner.Bot.Azure.Factories;
 
 internal class TelegramBotClientFactory : ITelegramBotClientFactory {
+    private readonly IApplicationSettings ApplicationSettings;
```
Comment (PublicFedorenko) [inline FaynaTown.Tennis.Planner.Bot.Azure/Factories/TelegramBotClientFactory.cs:9]:
```suggestion

```

File: FaynaTown.Tennis.Planner.Bot.Azure/Factories/TelegramBotClientFactory.cs [hunk 5]
```diff
@@ -1,17 +1,20 @@
...
     private readonly ISecretProvider SecretProvider;
-    private TelegramBotClient? TelegramBotClient;
+    private ITelegramBotClient? TelegramBotClient;
 
-    public TelegramBotClientFactory(ISecretProvider secretProvider) {
+    public TelegramBotClientFactory(IApplicationSettings applicationSettings, ISecretProvider secretProvider) {
```
Comment (PublicFedorenko) [inline FaynaTown.Tennis.Planner.Bot.Azure/Factories/TelegramBotClientFactory.cs:13]:
```suggestion
    public TelegramBotClientFactory(ISecretProvider secretProvider) {
```

File: FaynaTown.Tennis.Planner.Bot.Azure/Factories/TelegramBotClientFactory.cs [hunk 6]
```diff
@@ -1,17 +1,20 @@
...
         SecretProvider = Guard.Against.Null(secretProvider);
+        ApplicationSettings = Guard.Against.Null(applicationSettings);
```
Comment (PublicFedorenko) [inline FaynaTown.Tennis.Planner.Bot.Azure/Factories/TelegramBotClientFactory.cs:15]:
```suggestion

```

**Review by PublicFedorenko — CHANGES_REQUESTED** [review]

**Pull request merged**